src/sbvr-api/uri-parser.ts

```diff
@@ -0,0 +1,225 @@
+import * as _AbstractSQLCompiler from '@resin/abstract-sql-compiler'
+
+import * as Promise from 'bluebird'
+import TypedError = require('typed-error')
+const { ODataParser } = require('@resin/odata-parser')
+interface OData2AbstractSQLInstance {
+	match: (abstractSQL: ODataQuery, rule: 'Process', args: [SupportedMethod, string[]]) => {
+		tree: _AbstractSQLCompiler.AbstractSqlQuery,
+		extraBodyVars: {},
+	}
+	setClientModel: (abstractSqlModel: _AbstractSQLCompiler.AbstractSqlModel) => void
+}
+const { OData2AbstractSQL }: {
+	OData2AbstractSQL: {
+		createInstance: () => OData2AbstractSQLInstance
+	}
+} = require('@resin/odata-to-abstract-sql')
+import * as memoize from 'memoizee'
+import * as _ from 'lodash'
+
+export class TranslationError extends TypedError {}
+export class ParsingError extends TypedError {}
+export class BadRequestError extends TypedError {}
+
+type SupportedMethod = "GET" | "PUT" | "POST" | "PATCH" | "MERGE" | "DELETE" | "OPTIONS"
+type ODataQuery = any[]
+
+export interface UnparsedRequest {
+	method: string
+	url: string
+	data: any
+	headers?: { [header: string]: string }
```

Comment:
I think this should be `{ [header: string]: string[] }` 
(Im not sure why header values are list of strings instead of simple strings, but this comes from dicer which is the multipart request parser)